src/Standards/Generic/Docs/Formatting/SpaceBeforeCastStandard.xml

```diff
@@ -0,0 +1,31 @@
+<documentation title="Space Before Casts">
+    <standard>
+    <![CDATA[
+    Ensures there is exactly one space before a cast operator.
+    ]]>
+    </standard>
+    <code_comparison>
+        <code title="Valid: Single space before a cast operator.">
+        <![CDATA[
+$integer =<em> </em>(int) $string;
+        ]]>
+        </code>
+        <code title="Invalid: No space before a cast operator.">
+        <![CDATA[
+$integer =<em></em>(int) $string;
+        ]]>
+        </code>
+    </code_comparison>
+    <code_comparison>
+        <code title="Valid: Single space before a cast operator.">
+        <![CDATA[
+$c = $a +<em> </em>(int) $b;
+        ]]>
+        </code>
+        <code title="Invalid: Multiple spaces before a cast operator.">
+        <![CDATA[
+$c = $a +<em>   </em>(int) $b;
+        ]]>
+        </code>
```

Comment:
As the valid code is essentially the same for both cases, I think it might be better to use a single `<code_comparison>` block, where the left side would have either just the one or both of the "valid" examples and the right side the two "invalid" examples.

Might also be good to vary the type casts used in the code samples a little as people _could_ get the impression this is only about `(int)` type casts based on the current cod samples.

Reply:
Even though the valid code is essentially the same for both cases, I opted for this approach as it was the only way that I could find to have titles for the two different invalid codes to highlight that this sniff acts when there is no space and also when there are multiple spaces.

> I think it might be better to use a single <code_comparison> block, where the left side would have either just the one or both of the "valid" examples and the right side the two "invalid" examples.

I might be missing something, but I was not able to find a way to have more than two `<code>` blocks inside a `<code_comparison>` block. Or is your suggestion that I combine both invalid examples in a single `<code>` block?

> Might also be good to vary the type casts used in the code samples a little as people could get the impression this is only about (int) type casts based on the current code samples.

:+1: 

Maybe the second example could be:

```
        <code title="Valid: Single space before a cast operator.">
        <![CDATA[
$c = $a . <em> </em>(string) $b;
        ]]>
        </code>
        <code title="Invalid: Multiple spaces before a cast operator.">
        <![CDATA[
$c = $a .<em>   </em>(string) $b;
        ]]>
        </code>
```

(I'm happy to adjust the example above and include it in the first `<code_comparison>` block depending on what you suggest as the best approach for the other topic that we are discussing in this thread)

Reply:
> If it clarifies things, you can use inline comments in the code sample to add some additional info, like so:

It's good to know that this is an option. Thanks.

> Having said that, I don't really think those comments are needed in this case as "no space" and "too much space" both translate to "not using a single space", so a good description for the "invalid" case can cover both.

I just pushed 1f90b17561cd0581ab283bb1db0cb7da9f3054ad in which I combined both code examples into a single `<code_comparison>`, changed the second code example to use `(string)` instead of `(int)` and updated the title of the invalid case to mention both no space and multiple spaces.

>> what you suggest as the best approach for the other topic that we are discussing in this thread

> Which other topic ?

Our conversation on whether to use one or two `<code_comparison>` blocks. I just wanted to make it explicit that with the example that I provided, I was not implying that we should continue to use two `<code_comparison>` blocks.